README.md

```diff
@@ -35,6 +30,30 @@ The column is calculated by the operation i % 14. This will also return an integ
 Using both calculations, we can now combine them to get the following equation.
 mapStringMatrix[i/14][i%14] = mapString[i]
 
+## Path finding
+
+For path finding, we used a simpler version of Dijstra's algorithm with lower time complexity, so that the robot can only move in four directions. The algorithm doesn't need to check every other node for a possible shorter path, but only the four adjacent tiles. 
+
+All the information for the pathfinding algorithm is contained in an array of the Structure tile. A tile contains the information required for one tile of the map, which is referred to as a node in this context.
+The structure tiles contains the following information:
+- visited | has the tile been processed by the algorithm (bool)
+- distance | the current shortest distance to the starting position (uint8_t)
+- prev | the index of the previous tile on the current shortest path as 1D index (uint8_t)
+
+In order to save memory, all indexes are stored as 1D indexes. We have written a function to convert these coordinates into 2D indexes and vice versa. Hence to this we can always use the index that's more convenient for the current calculation.
+
+The are parts of the algoritm are pretty much equal to a "normal" Dijkstra. In order to explain other interesting logic in more detail, a detailed explanation the algorithm is omitted.
+
+## Conversion to driving instructions 
+
+To find the shortest path, we first look for the neighbouring tile of the current destination with the shortest distance to the start. Then a route array of that particular distance is constructed. This array contains the coordinates of the tiles from the start (excluded) to the destination (included). This array is used to generate the driving instructions, with the forward function being executed only if a turn is required or if there are no more coordinates to process.
+
+When the robot arrives at the destination, the makeSound() function is called and the robot turns 180 degrees.
+
+For the way back, a similar array is constructed, now containing the coordinates of the start tile. The execution of the motion commands is done in the same way, the only difference being that the array is now processed from the back to the front.  
```

Comment:
- The execution of the motion commands 'are' done in the ...

Reply:
is should be fine here since its refering to the execution not the motion commands